worker-taskunstowing/src/main/java/com/github/jobservice/workers/taskunstowing/TaskUnstowingWorker.java

```diff
@@ -0,0 +1,250 @@
+                    stowedTaskRowsToExclude.add(stowedTaskRow); // Exclude task from subsequent iterations of while loop.
+                    final TaskUnstowingWorkerFailure failure
+                        = TaskUnstowingWorkerFailure.FAILED_TO_SEND_UNSTOWED_TASK_MESSAGE_TO_QUEUE_ID;
+                    processFailure(failure, exception, document, partitionId, jobId, taskMessage.getTracking().getJobTaskId(),
+                                   taskMessage.getTo());
+                    continue;
+                }
+
+                try {
+                    databaseClient.deleteStowedTask(partitionId, jobId, taskMessage.getTracking().getJobTaskId());
+                    LOGGER.info("Deleted stowed task with partitionId={}, jobId={} and jobTaskId={} from database",
+                                partitionId, jobId, taskMessage.getTracking().getJobTaskId());
+                } catch (final Exception exception) {
+                    stowedTaskRowsToExclude.add(stowedTaskRow); // Exclude task from subsequent iterations of while loop.
+                    final TaskUnstowingWorkerFailure failure
+                        = TaskUnstowingWorkerFailure.FAILED_TO_DELETE_UNSTOWED_TASK_MESSAGE_FROM_DATABASE_ID;
+                    if (DatabaseExceptionChecker.isTransientException(exception)) {
+                        LOGGER.error(failure.toString(
+                            partitionId, jobId, taskMessage.getTracking().getJobTaskId(), taskMessage.getTo()), exception);
+                        throw new DocumentWorkerTransientException(failure.toString(
+                            partitionId, jobId, taskMessage.getTracking().getJobTaskId(), taskMessage.getTo()), exception);
+                    } else {
+                        processFailure(failure, exception, document, partitionId, jobId, taskMessage.getTracking().getJobTaskId(),
+                                       taskMessage.getTo());
+                    }
+                }
+            }
+
+            // Fetch next batch of stowed tasks.
+            LOGGER.info(
+                "Fetching next batch (using max batch size {}) of stowed task(s) for partitionId={} and jobId={} from database",
+                databaseMaximumBatchSize, partitionId, jobId);
+            try {
+                stowedTaskRows = databaseClient.getStowedTasks(partitionId, jobId, databaseMaximumBatchSize);
+                LOGGER.info("Fetched {} stowed task(s) for partitionId={} and jobId={} from database",
+                            stowedTaskRows.size(), partitionId, jobId);
+
+                // Exclude any rows that were failed to be processed from the next batch so we don't get stuck in an infinite loop.
+                final int numStowedTaskRowsBeforeExcluding = stowedTaskRows.size();
+                stowedTaskRows = stowedTaskRows.stream()
+                    .filter(stowedTask -> !stowedTaskRowsToExclude.contains(stowedTask))
+                    .collect(Collectors.toList());
```

Comment:
This may have been more efficient in SQL but I couldn't figure out exactly how to do it. I think it's probably ok as this is an error case scenario and hopefully unlikely to happen often.